src/main/java/org/opensearch/flowframework/util/EncryptorUtils.java

```diff
@@ -204,9 +204,29 @@ String decrypt(final String encryptedCredential) {
      * @param template the template
      * @return the redacted template
      */
-    public Template redactTemplateCredentials(Template template) {
+    public Template redactTemplateSecuredFields(Template template) {
+        Template updatedTemplate = null;
+
+        if (template.getUser() != null) {
+            updatedTemplate = new Template.Builder(template).name(template.name())
+                .description(template.description())
+                .useCase(template.useCase())
+                .templateVersion(template.templateVersion())
+                .user(null)
+                .uiMetadata(template.getUiMetadata())
+                .compatibilityVersion(template.compatibilityVersion())
+                .workflows(template.workflows())
+                .createdTime(template.createdTime())
+                .lastUpdatedTime(template.lastUpdatedTime())
+                .lastProvisionedTime(template.lastProvisionedTime())
+                .build();
```

Comment:
The Builder constructor using the old template already copies over the other fields
```suggestion
            updatedTemplate = new Template.Builder(template)
                .user(null)
                .build();
```

Reply:
Ah! I missed to change this after the testing